lib/semian/simple_sliding_window.rb

```diff
@@ -0,0 +1,45 @@
+module Semian
+  module Simple
+    class SlidingWindow #:nodoc:
+      extend Forwardable
+
+      def_delegators :@window, :size, :pop, :shift, :first, :last
+      attr_reader :max_size
+
+      def initialize(max_size:)
+        @max_size = max_size
+        @window = []
+      end
+
+      # A sliding window is a structure that stores the most @max_size recent timestamps
+      # like this: if @max_size = 4, current time is 10, @window =[5,7,9,10].
+      # Another push of (11) at 11 sec would make @window [7,9,10,11], shifting off 5.
+
+      def resize_to(size)
+        raise ArgumentError.new('size must be larger than 0') if size < 1
+        @max_size = size
+        @window.shift while @window.size > @max_size
+        self
+      end
```

Comment:
Since this is only used in tests, I don't think you should make it part of the public interface.